crates/iota/src/name_commands.rs

```diff
@@ -1028,8 +1029,8 @@ impl AuctionCommand {
 #[derive(Parser)]
 #[command(rename_all = "kebab-case")]
 pub enum SubnameCommand {
-    /// Register a new leaf subname, which can only be managed by the parent's
-    /// NFT
+    /// Register a new leaf subname, which will NOT create an NFT but instead
+    /// is managed by its parent NFT. Therefore it won't be listed by the CLI.
```

Comment:
I tried to explain a bit more in depth without being overly verbose. I think clap will only display the first line by default, and additional lines with a special flag, but I'm not sure.

```suggestion
    /// Register a new leaf subname.
    /// Leaf subnames have no associated NFT; their ownership is determined
    /// by the ownership of the parent's NFT. Unfortunately, this makes
    /// it difficult to find leafs owned by a particular address. For this
    /// reason, leaf subnames are not included in the `list` command.
```

Reply:
And if we like this then I propose we add it to the `list` command too, slightly reworded

Reply:
I would prefer your text to be part of a separate CLI documention, that goes more into the "why" question. In order to be as concise as possible, I perdonally would prefer for the CLI to restrict itself to answer the "what" question.

Reply:
@thibault-martinez @Thoralf-M what do you think?

Reply:
@thibault-martinez @Thoralf-M what do you think?

Reply:
> to be part of a separate CLI documention

Where would this be? If it's not here, then it will probably become outdated quickly and also won't help people that only use --help

Reply:
In `cli.mdx` there has to be a reference to a new `./cli/iota_names.mdx` document, where this can be explained in detail. I personally don't think that phrases like "Unfortunately..." belong in the CLI, this belongs to an actual document. But that's just my opinion!! 

Why would it get outdated quickly? Same could be said for the other CLI documents then. No?

Reply:
Yes and the issue is that they are already outdated right now :cry: Because it's not part of the code and we don't update the docs for every little change we do here

Reply:
At some point the CLI will stabilize, and docs won't be outdated that quickly.

Reply:
I agree with Alex that here I would just explain the "what" and not the "why. Remember that we will have proper IOTA Names documentation and we can expand the "why" there.

Reply:
Okay, but the what is not useful here because it doesn't make sense without the why

Reply:
I am still not convinced that the CLI is actually the right place for the user to understand WHY a command works in a specific way. Disadvantages I see are:
* overly verbose CLI docs leading to a messy terminal view
* leaking implementation details
* quicker desynchronization between code and docs

Reply:
> Okay, but the what is not useful here because it doesn't make sense without the why

I completely disagree, we can definitely have the what without the why, and I think we should

Reply:
after some discussion we decided to just state the facts: https://github.com/iotaledger/iota/pull/7845/commits/ae91c7642e0983a2c0143208e22cf4135e6c3143